### PR TITLE
Do not create or block on api review for nested packages

### PR DESCRIPTION
**Pull request by benbp**
I do not think there are any pipelines where we need/want to be recursing on packages, as we keep to a one pipeline per package convention in go data plane. Currently nested packages, i.e. azidentity/cache are blocking parent packages from releasing if they don't have an api review approval.

### CODE REVIEW

File: eng/scripts/Create-ApiReview.ps1
Comment (weshaggard) [inline eng/scripts/Create-ApiReview.ps1:27]: I don't have enough context to know if there is a better option but this seems to have a smell to me. Should we be filtering at this level or at the Get-AllPackageInfoFromRepo level, or perhaps some other pipeline filter configuration?

Comment (benbp) [reply eng/scripts/Create-ApiReview.ps1:27]: This is a bit of a quick fix to unblock identity release. I'm going to be looking at a broader recurse/norecurse option at the higher level because it seems that both are used in different places.

**Review by chlowell — APPROVED** [review]

**Review by RickWinter — CHANGES_REQUESTED** [review]
How are we verifying the sub-packages have APIView signoff?

**Comment by chlowell** [comment]
It's perhaps confusing here that EngSys scripts like this one often reference "packages" where they mean modules. We create API reviews for modules, not packages, so e.g. `azcore/runtime` appears in the review for `azcore` and will continue to do so with this change because the pipeline will still run apiview automation in `skd/azcore`. What we don't want is to run that automation in a subdirectory of the module root that also contains a `go.mod` i.e., is a separate module. The script does that today because it recursively discovers ~~packages~~ modules.

**Pull request merged**
